src/chainparams.h

```diff
@@ -14,6 +14,8 @@
 
 #include <vector>
 
+#include <rust/wallet_scanner.h>
```

Comment:
Is this an appropriate dependency? Does it still build if ENABLE_WALLET is not set?

Reply:
`ENABLE_WALLET` currently has no effect on the Rust side. That's not something we've spent time to integrate yet. So it is already the case that extra code is compiled on the Rust side, though in theory LTO should be removing it as we statically link everything.

This dependency here is something we need due to a limitation in `cxx` (we cannot access Rust bridge types in other Rust bridges). It should be fine for the above reason, though we should also confirm that `libzcash_script` isn't affected badly.

Reply:
Are you sure? The last part of https://github.com/dtolnay/cxx/issues/353 hasn't been implemented, but https://github.com/dtolnay/cxx/issues/297#issuecomment-727042059 seems to say you can work around it. Naively I would have assumed that you can put the code in the `#[namespace = "consensus"]` [block in wallet_scanner.rs](https://github.com/zcash/zcash/pull/6474/commits/a95cd3db09112b7b76314155f36fbc453fd7983c#diff-41bdfef39b0d61583916ccfd773d45ac7590187eaa082edce57b238b12233909R40-R53) into `src/rust/src/params.rs`, and then use the pattern in that comment to import it from wallet_scanner &mdash; which I think would better reflect the logical dependencies.

Reply:
That is showing how to share C++ types defined across different bridges, which is [documented here](https://cxx.rs/extern-c++.html#reusing-existing-binding-types). Rust types cannot yet be shared across different bridges.

Reply:
And indeed, splitting the bridges up into separate files and then having separate C++ imports is where I'd like to go, but that is blocked on `cxx`.